spec/api/keys/keys_spec.rb

```diff
@@ -944,23 +1050,48 @@ def make_user_payload(payload)
           delete("#{platform.server}/users/#{other_org_user_name}", superuser).should look_like({:status => 200})
         end
 
-        it "for a client that is a member of the same org fails with a 403" do
-          list_client_keys($org['name'], client['name'], org_client).should look_like({:status => 403})
-        end
-        it "for a itself succeeds with a 200" do
-          list_client_keys($org['name'], org_client_name, org_client).should look_like({:status => 200})
+        context "when GET /organizations/org/clients/client/keys is called" do
+          it "for a client that is a member of the same org fails with a 403" do
+            list_client_keys($org['name'], client['name'], org_client).should look_like({:status => 403})
+          end
+          it "for a itself succeeds with a 200" do
+            list_client_keys($org['name'], org_client_name, org_client).should look_like({:status => 200})
+          end
+          it "for a client that is a member of a different org fails with a 401" do
+            list_client_keys($other_org_name, other_org_client_name, org_client).should look_like({:status => 401})
+          end
         end
 
-        it "for a client that is a member of a different org fails with a 401" do
-          list_client_keys($other_org_name, other_org_client_name, org_client).should look_like({:status => 401})
+        context "when GET /users/user/keys is called" do
+          it "for a user that is a member of the same org fails with a 401" do
+            list_user_keys(org_user_name, org_client).should look_like({:status => 401})
```

Comment:
I remember now, I think - the request is handled outside the context of the org, so the org client is unauthorized and we don't get as far as access checks.